src/domain/safe/safe.repository.ts

```diff
@@ -95,6 +96,18 @@ export class SafeRepository implements ISafeRepository {
     return page;
   }
 
+  async addConfirmation(
+    chainId: string,
+    safeTxHash: string,
+    addConfirmationDto: AddConfirmationDto,
+  ): Promise<void> {
+    const transactionService =
+      await this.transactionApiManager.getTransactionApi(chainId);
+    await transactionService.postConfirmation(safeTxHash, addConfirmationDto);
+    const transaction = await this.getMultiSigTransaction(chainId, safeTxHash);
```

Comment:
This is something that we should consider in the future. We are doing an HTTP request just to get the Safe associated with this transaction.

This might not be a big issue depending on the cache timeouts that we have in place at the moment.